### ISSUE TITLE
[Bug]: Accessible names for sp-menu-item are not showing up in Blink-based browsers

### ISSUE THREAD

**Issue by astearns** [adobe/spectrum-web-components#3931]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

sp-menu-item

### Expected behavior

If I inspect the accesssibility properties of an sp-menu-item (like the long one in https://opensource.adobe.com/spectrum-web-components/components/menu-item/#value-attribute) I should see an accessible name on an option set to the contents. This works for me in Mac Firefox, but not Mac Edge/Chrome

### Actual behavior

In the accessibility tree an 'option' appears with no accessible name

### Screenshots

Good Firefox:
![Screenshot 2024-01-09 at 3 27 56 PM](https://github.com/adobe/spectrum-web-components/assets/1380751/5813e3e4-5e9b-431d-84aa-cefbd0d74fe0)

Bad Edge:
![Screenshot 2024-01-09 at 3 28 38 PM](https://github.com/adobe/spectrum-web-components/assets/1380751/af2812ea-6236-4a2a-84b1-b4e1ba0616bc)

### What browsers are you seeing the problem in?

Chrome, Microsoft Edge

### How can we reproduce this issue?

1. Go to https://opensource.adobe.com/spectrum-web-components/components/menu-item/#value-attribute
2. Click on the Menu Items Examples picker
3. Right-click on the long option and choose inspect
4. Choose the accessibility tab
5. Look for the 'option' inside the listbox and see the blank area where the accessible name should be

(this also happens for the 'menuitem's inside of the sp-action-menu in the example to the right)

**Comment by najikahalsema** [comment]
@astearns I would posit that this is because the `value` attribute doesn't actually have anything to do with setting the `aria-label` for an `sp-menu-item`. You should instead use `label` to get the accessibility benefits, but that is not clear at all from the documentation! The reason for this is that `MenuItem` is extended from another base class, `LikeAnchor`, which has a `label` attribute, but isn't even on Menu Item's API table... so, yikes. 

In fact, I think it makes sense for the `aria-label` (and potentially, the `label` attribute itself) of the menu item to be set with the `value` if it exists and the `label` attribute has not already been explicitly set by the user. 

What do you think, @Westbrook? Is it worth using the `value` attribute to update the `aria-label` if it interacts well with the screen reader?

**Comment by najikahalsema** [comment]
Okay, scratch everything I just said. Could you create an example of this not working that's similar to where you encountered this in your local environment by [forking this example](https://studio.webcomponents.dev/edit/collection/fO75441E1Q5ZlI0e9pgq/guTpKeAjgecibF150Qbg/src/index.ts?p=stories)? I'm able to see the accessibility tree on this example on Chrome. I can't see the accessbility tree for the docs site menu you referenced, though, which is strange.

**Comment by astearns** [comment]
With no forking, I am not seeing an accessible name for the first option in Mac Edge:

<img width="992" alt="image" src="https://github.com/adobe/spectrum-web-components/assets/1380751/937bce96-5888-4322-8c14-e60546bfe0fb">

**Comment by najikahalsema** [comment]
Hi, we did a release this morning, and version 0.40.3 seems to fix this issue on both Chrome and Edge via [this PR](https://github.com/adobe/spectrum-web-components/commit/fc0278b917759ed58c3ac62a6e962633914481c0). `visibility: hidden` was preventing elements in overlays from populating the a11y tree, even after they became visible.

I tested on Chrome and Edge, so if you want to check for yourself, please do and let me know if we can close this issue.

**Comment by najikahalsema** [comment]
@astearns Any update on whether this looks good for you so we can close this issue?

**Comment by astearns** [comment]
It looks fixed on https://opensource.adobe.com/spectrum-web-components/components/menu-item/#value-attribute but I am getting mixed results on [your example](https://studio.webcomponents.dev/edit/collection/fO75441E1Q5ZlI0e9pgq/guTpKeAjgecibF150Qbg/src/index.ts?p=stories) (with the branch set to master - does that have the 0.40.3 changes?)

**Comment by astearns** [comment]
I just retested and am getting good results for both links now. Thanks!
